scene/2d/audio_stream_player_2d.h

```diff
@@ -52,6 +52,7 @@ class AudioStreamPlayer2D : public Node2D {
 	};
 
 	Vector<Ref<AudioStreamPlayback>> stream_playbacks;
+	HashMap<Ref<AudioStreamPlayback>, int> stream_looped;
```

Comment:
It is obviously hackey to have a map of each stream just to subscribe to each loop. So you should implement a looped flag in the AudioStreamPlayback, then connect and disconnect to the AudioStreamPlayer correctly and then fire the signal there.

Reply:
Should the looped flag be reset when it's read by the playback or by the player when it's read? I suppose it's probably a better idea to auto-reset it, but also then why does get_loop_count() even exist? It's not actually available from GDScript, despite being in the docs you cannot call it.

Reply:
Actually, if you meant "signal" by "flag" then this is probably a better solution, yes.

Reply:
I'm not sure what you mean by reset. It just needs to fire a signal from AudioStreamPlayback at the moment of loop, i.e. the moment the `loops` increments in the AudioStreamPlayback child class. Then subscribes to it in AudioStreamPlayer, and then fires more signal from AudioStreamPlayback. Signals must be connected and disconnected at the time of loading and deleting the streams.

Reply:
> So you should implement a looped flag

 Oh, sorry, it was signal, not flag.

Reply:
Re-implemented this with signaling and tracking the looped on the players with `CONNECT_REFERENCE_COUNTED`, since this seems to be one of the cases where this is what we want.

I hope my implementation of the `looped` signal on the randomizer is OK, I can think of a use case for this signal on it so I thought I'd add it. The playbacks that I didn't add the signal to do not support looping and always return 0 for the count of loops.